lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -736,6 +739,93 @@ def upload_students_csv(_xmodule_instance_args, _entry_id, course_id, task_input
+
+    fmt = u'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
+    task_info_string = fmt.format(
+        task_id=_xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None,
+        entry_id=_entry_id,
+        course_id=course_id,
+        task_input=_task_input
+    )
+    TASK_LOG.info(u'%s, Task type: %s, Starting task execution', task_info_string, action_name)
+
+    # Loop over all our students and build our CSV lists in memory
+    rows = []
+    header = None
+    current_step = {'step': 'Gathering Profile Information'}
+    enrollment_report_provider = CyberSourceEnrollmentReportProvider()
+    total_students = students_in_course.count()
+    student_counter = 0
+    TASK_LOG.info(
+        u'%s, Task type: %s, Current step: %s, Starting grade calculation for total students: %s',
+        task_info_string,
+        action_name,
+        current_step,
+        total_students
+    )
+
+    for student in students_in_course:
+        # Periodically update task status (this is a cache write)
+        if task_progress.attempted % status_interval == 0:
+            task_progress.update_task_state(extra_meta=current_step)
+        task_progress.attempted += 1
+
+        # Now add a log entry after certain intervals to get a hint that task is in progress
+        student_counter += 1
+        if student_counter % 100 == 0:
+            TASK_LOG.info(
+                u'%s, Task type: %s, Current step: %s, gathering enrollment profile for students in progress: %s/%s',
+                task_info_string,
+                action_name,
+                current_step,
+                student_counter,
+                total_students
+            )
+
+        user_data = enrollment_report_provider.get_user_profile(student.id)
+        course_enrollment_data = enrollment_report_provider.get_enrollment_info(student, course_id)
+        payment_data = enrollment_report_provider.get_payment_info(student, course_id)
+
+        if not header:
+            header = user_data.keys() + course_enrollment_data.keys() + payment_data.keys()
+            rows.append(header)
+
+        rows.append(user_data.values() + course_enrollment_data.values() + payment_data.values())
+        task_progress.succeeded += 1
+
+    TASK_LOG.info(
+        u'%s, Task type: %s, Current step: %s, Grade calculation completed for students: %s/%s',
+        task_info_string,
+        action_name,
+        current_step,
+        student_counter,
+        total_students
+    )
+
+    # By this point, we've got the rows we're going to stuff into our CSV files.
+    current_step = {'step': 'Uploading CSVs'}
+    task_progress.update_task_state(extra_meta=current_step)
+    TASK_LOG.info(u'%s, Task type: %s, Current step: %s', task_info_string, action_name, current_step)
+
+    # Perform the actual upload
+    upload_csv_to_report_store(rows, 'enrollment_report', course_id, start_date)
```

Comment:
We'll need to add another parameter here and wire it through to the instantiation of ReportStore, so we can ultimately call down into ReportStore to select different configuration than the GRADES_DOWNLOAD. This is because of permissioning: users who can download grade reports will not necessarily have permissions to download financial reports, so we need to have separate configuration to specify different S3 buckets. Thanks